Source/cmake/OptionsGTK.cmake

```diff
@@ -51,6 +51,7 @@ WEBKIT_OPTION_DEFINE(USE_LIBDRM "Whether to enable usage of libdrm." PUBLIC ON)
 WEBKIT_OPTION_DEFINE(USE_LIBHYPHEN "Whether to enable the default automatic hyphenation implementation." PUBLIC ON)
 WEBKIT_OPTION_DEFINE(USE_LIBSECRET "Whether to enable the persistent credential storage using libsecret." PUBLIC ON)
 WEBKIT_OPTION_DEFINE(USE_SOUP2 "Whether to enable usage of Soup 2 instead of Soup 3." PUBLIC OFF)
+WEBKIT_OPTION_DEFINE(ENABLE_JSC_RESTRICTED_OPTIONS_BY_DEFAULT "Whether to enable dangerous development options in JSC by default." PUBLIC OFF)
```

Comment:
Looks like this is unused leftover and can be removed?

(But if it was not unused, I would next ask why it should be public if it's a dangerous development option, and also why it needs to be specific to GTK port instead of going into WebKitFeatures.cmake.)

Reply:
The Apple ports enable this for debug builds automatically now, so it really is specific to our ports. Really, it should be on by default for development builds, and off everywhere else. What is the best way to accomplish that?

Reply:
Technically, it's on for all development builds (i.e. Release/Debug but not Production/Profiling) on Apple ports. Not sure if that's particularly important but I thought I'd mention it.

Reply:
> The Apple ports enable this for debug builds automatically now

No they don't. A `git grep` indicates `ENABLE_JSC_RESTRICTED_OPTIONS_BY_DEFAULT` does not exist. :)

> Really, it should be on by default for development builds, and off everywhere else. What is the best way to accomplish that?

In that case, doing it in OptionsGTK.cmake and OptionsWPE.cmake really is required, because `DEVELOPER_MODE` is a platform-specific option. But please don't make it `PUBLIC`.

Reply:
> > The Apple ports enable this for debug builds automatically now
> 
> No they don't. A `git grep` indicates `ENABLE_JSC_RESTRICTED_OPTIONS_BY_DEFAULT` does not exist. :)

Yeah, it's there, in `Source/JavaScriptCore/Configurations/DebugRelease.xcconfig`:
```
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) ENABLE_JSC_RESTRICTED_OPTIONS_BY_DEFAULT=1;
```